### ISSUE TITLE
SQL WMI .Services and .ServerInstances enumeration + permissions

### ISSUE THREAD

**Issue by potatoqualitee** [microsoft/sqlmanagementobjects#101]
Greets, I'm having issues securing a machine -- particularly setting file permissions. When I remove Inheritance from `C:\Program Files\Microsoft SQL Server` then remove the `COMPUTER\Users` user and apply to all children, I run into enumeration issues when I executing

(New-Object Microsoft.SqlServer.Managmeent.Smo.WMI.ManagedComputer).ServerInstances (or .Services). I receive the error "The following exception occurred while trying to enumerate the collection: "An exception occurred while trying to manage a service".

This occurs in powershell.exe running as administrator, as required. The user I'm running it as is an administrator which has full access to the folder.

Any idea what's going on? I see this issue randomly when I secure servers. In this case, I'm running SQL Server 2016 -- 13.0.6419.1

**Comment by shueybubbles** [comment]
have you tried using some other WMI tool like WBEMTEST to do the same thing? It sounds like the WMI provider is falling over.  I suspect WBEMTEST would show the same problem.

**Comment by potatoqualitee** [comment]
Thanks, just tried it and received an ACCESS DENIED when running `select * from sqlservice`. I checked and the `Windows Management Instrumentation` and it runs as Local System which has access.

I returned to the text and realized that `C:\Program Files\Microsoft SQL Server` is simply too high up the tree to change the permissions -- the randomness is probably me securing the appropriate directory and attempting to secure the whole directory. I still don't know why it doesn't work, but I don't have to now haha! Changing the MSSQL directory is all I needed and WMI continues to work. Thanks agin.
